File: functions/api/src/routes/format.rs
use lambda_runtime::Error;
use serde::Deserialize;
use serde_json::{Value, json};
use wipple_compiler::Compiler;

#[derive(Deserialize)]
#[serde(deny_unknown_fields, rename_all = "camelCase")]
pub struct FormatRequest {
    code: String,
}

pub async fn handle(req: FormatRequest) -> Result<Value, Error> {
    Ok(json!({
        "code": Compiler::format(&req.code),
    }))
}

File: server/src/routes/format.ts
import * as z from "zod";
import compiler from "compiler";
import { handler } from "../handler";

const FormatRequest = z.object({
    code: z.string(),
});

export default handler(FormatRequest, async (body) => {
    const formatted = compiler.format(body.code);

    return {
        statusCode: 200,
        body: { code: formatted != null && formatted !== body.code ? formatted : null },
    };
});
